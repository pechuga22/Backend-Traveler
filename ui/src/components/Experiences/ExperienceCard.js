import React from 'react'
import { Comment, Icon, Rating, Image } from 'semantic-ui-react'
import moment from 'moment'

import './ExperienceCard.less'
import Resources from '../Resources/Resources'

export default function ExperienceCard({ experience }) {
  moment.locale('en', {
    relativeTime: {
      future: 'in %s',
      past: '%s ago',
      s: 's',
      m: '1 min',
      mm: '%d min',
      h: '1 h',
      hh: '%d h',
      d: '1 d',
      dd: '%d d',
      M: '1 m',
      MM: '%d m',
      y: '1 y',
      yy: '%d y',
    },
  })
  const {
    _id,
    avatar,
    name,
    creationDate,
    rating,
    content,
    resources,
  } = experience
  return (
    <Comment.Group className='recomendations'>
      <Comment id={_id}>
        <Image
          src={avatar}
          floated='left'
          circular
          rounded
          size='mini'
          avatar
          bordered
        />
        <Comment.Content>
          <Comment.Author as='a'>{name}</Comment.Author>
          <Comment.Metadata>
            <Icon className='clock outline' />
            <div>{moment(creationDate).fromNow()}</div>
          </Comment.Metadata>
          <div>
            <Rating icon='favorite' defaultRating={rating} maxRating={rating} />
          </div>
        </Comment.Content>
        <Comment.Text>{content}</Comment.Text>
        {resources ? <Resources resources={resources} /> : null}
      </Comment>
    </Comment.Group>
  )
}
