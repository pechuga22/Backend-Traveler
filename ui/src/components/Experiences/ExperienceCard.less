@import '../../variables.less';

.ui.recomendations {
  .comment {
    a.author {
      color: @primaryColor;
      font-size: 16px;
      text-transform: capitalize;
    }

    div.text {
      color: @secondaryColor;
      margin-top: 1.5rem;
    }

    .ui.bordered.circular.left.floated.image.avatar {
      height: 3.5rem;
      width: 3.5rem;
      object-fit: contain;
      border-radius: 50%;
      margin-right: 0.5rem;
    }

    div.metadata {
      float: right;
    }

    div.text {
      margin-top: 0.5rem;
    }
  }
}
