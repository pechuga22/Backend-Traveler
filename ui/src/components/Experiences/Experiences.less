@import '../../variables';

.experiences.tour {
  margin-bottom: 110px;
  .large {
    font-weight: bold;
  }

  .experiences {
    margin-top: 0;
  }

  .see.experiences {
    text-align: center;
  }

  .ui.button.btn.experiences {
    font-size: 12px;
    background: none;
  }
}
