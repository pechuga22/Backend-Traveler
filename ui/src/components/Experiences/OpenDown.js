import React from 'react'
import { Button } from 'semantic-ui-react'

export default function OpenDown({
  getAllExperiences,
  experiences,
  totalExperiences,
}) {
  const openDown =
    experiences.length !== totalExperiences ? (
      <Button onClick={getAllExperiences} className='btn experiences'>
        SEE ALL EXPERIENCES
      </Button>
    ) : (
      <Button disabled className='btn experiences'>
        THERE ARE NOT MORE EXPERIENCES
      </Button>
    )
  return <div className='see experiences'>{openDown}</div>
}
