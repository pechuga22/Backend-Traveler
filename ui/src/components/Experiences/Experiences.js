import React, { useState, useEffect } from 'react'
import { Container, Header } from 'semantic-ui-react'
import useFetch from 'use-http'

import { TourServices } from '../../services'
import ExperienceCard from './ExperienceCard'
import OpenDown from './OpenDown'
import Loading from '../Loading/Loading'
import Error from '../Errors/Error'
import './Experiences.less'

export default function Experiences({ tourId }) {
  const [experiences, setExperiences] = useState([])
  const [totalExperiences, setTotalExperiences] = useState(0)
  const [page, setPage] = useState(1)
  const services = new TourServices()
  const { url } = services.experiences(tourId, page, 2)
  
  const { get, loading, error } = useFetch(url)

  async function getAllExperiences() {
    const getExperiences = await get()
    setExperiences(experiences.concat(getExperiences.data))
    setTotalExperiences(getExperiences.totalRows)
    setPage(page + 1)
  }

  useEffect(() => {
    getAllExperiences()
  }, [])

  return (
    <Container text className='experiences tour'>
      <Header as='h2' color='blue'>
        Travelers Experiences
      </Header>
      {loading && <Loading />}
      {error && <Error />}
      {experiences.map((experience) => (
        <ExperienceCard experience={experience} key={experience._id} />
      ))}

      <OpenDown
        getAllExperiences={getAllExperiences}
        experiences={experiences}
        totalExperiences={totalExperiences}
      />
    </Container>
  )
}
