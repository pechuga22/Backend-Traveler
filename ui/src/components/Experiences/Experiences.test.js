import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Experiences from './Experiences'

describe('./components/Experiences/Experiences.js', () => {
  describe('.render()', () => {
    const id = '5ec573d6f235707135952a94'

    it('should render the Experiences component', () => {
      const wrapper = shallow(<Experiences tourId={id} />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should have class experiences into Experiences component', () => {
      const component = shallow(<Experiences tourId={id} />)
      expect(component.find('.experiences').length).toEqual(1)
    })

    it('should not class into Experiences component', () => {
      const component = shallow(<Experiences tourId={id} />)
      expect(component.hasClass('/comment/')).toBe(false)
    })
  })
})
