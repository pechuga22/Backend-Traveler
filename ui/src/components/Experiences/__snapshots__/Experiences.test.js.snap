// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/Experiences/Experiences.js .render() should render the Experiences component 1`] = `
<Container
  className="experiences tour"
  text={true}
>
  <Header
    as="h2"
    color="blue"
  >
    Travelers Experiences
  </Header>
  <OpenDown
    experiences={Array []}
    getAllExperiences={[Function]}
    totalExperiences={0}
  />
</Container>
`;
